src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -114,76 +114,87 @@
 					@scroll="debounceHandleScroll">
 					<NcAppNavigationCaption :class="{'hidden-visually': !isSearching}"
 						:title="t('spreed', 'Conversations')" />
-					<Conversation v-for="item of conversationsList"
-						:key="item.id"
-						:ref="`conversation-${item.token}`"
-						:item="item" />
-					<template v-if="!initialisedConversations">
-						<LoadingPlaceholder type="conversations" />
+
+					<!-- Initial Loading -->
+					<LoadingPlaceholder v-if="!initialisedConversations" type="conversations" />
+
+					<!-- Conversations List -->
+					<template v-else-if="!isSearching">
+						<Conversation v-for="item of filteredConversationsList"
+							:key="item.id"
+							:ref="`conversation-${item.token}`"
+							:item="item" />
+						<Hint v-if="filteredConversationsList.length === 0" :hint="t('spreed', 'No matches')" />
 					</template>
-					<Hint v-else-if="noMatchFound"
-						:hint="t('spreed', 'No matches found')" />
-					<template v-if="isSearching">
-						<template v-if="!listedConversationsLoading && searchResultsListedConversations.length > 0">
+
+					<!-- Search results -->
+					<template v-else-if="isSearching">
+						<!-- Search results: user's conversations -->
+						<Conversation v-for="item of searchResultsConversationList"
+							:key="item.id"
+							:ref="`conversation-${item.token}`"
+							:item="item" />
+						<Hint v-if="searchResultsConversationList.length === 0" :hint="t('spreed', 'No matches')" />
+
+						<!-- Search results: listed (open) conversations -->
+						<template v-if="!listedConversationsLoading && searchResultsListedConversations.length !== 0">
 							<NcAppNavigationCaption :title="t('spreed', 'Open conversations')" />
 							<Conversation v-for="item of searchResultsListedConversations"
 								:key="item.id"
 								:item="item"
 								is-search-result />
 						</template>
+
+						<!-- Search results: users -->
 						<template v-if="searchResultsUsers.length !== 0">
 							<NcAppNavigationCaption :title="t('spreed', 'Users')" />
 							<NcListItem v-for="item of searchResultsUsers"
```

Comment:
Results are cached between different search attempts, until new response is received, maybe reset it?

```js
async fetchPossibleConversations() {
			this.contactsLoading = true
			this.searchResults = []
			this.searchResultsUsers = []
			this.searchResultsGroups = []
			this.searchResultsCircles = []
}
```
Or even move specific searchResults to computed properties?

Reply:
I'd make it in a separated from refactoring PR or discuss it in an issue as it changes behavior a bit. It makes the search "slower" from UX.